### DISCUSSION TITLE
Should we remove the "banners on SSL" flag/switch?

### DISCUSSION THREAD

**Discussion by drbyte** [zencart/zencart#6627] [Ideas]
Back before HTTPS/SSL was made commonplace, there was a need to be cautious about displaying banner images on pages that were served over https, to avoid "unsecure content" alerts being displayed by the browser, particularly during checkout, as that would drive conversions down because people sometimes abandoned their shopping cart purchase due to the confusion.

<img width="485" alt="Screen Shot 2024-08-03 at 9 59 31 PM" src="https://github.com/user-attachments/assets/8269b25c-2cae-4bdd-8d33-e658f799725e">

Now that all ecommerce stores are operated on https by default, we don't really need this configuration switch.

Can we remove it?

**Comment by lat9** [comment]
If it's going away, I'll need to remove its usage in the Bootstrap template: https://github.com/lat9/ZCA-Bootstrap-Template/blob/4a2a2c09e0644122844cbfe97d40e3813e2b8472/includes/templates/bootstrap/common/tpl_zca_banner_carousel.php#L9-L26

I have no clue whether other templates would also be affected.

**Reply by drbyte** [reply]
We could still define it by default, to `1` ("on"), for backward-compatibility.

**Reply by lat9** [reply]
That's a database field, not a constant, so removing that will result in SQL queries failing (if the field is specifically called out) or PHP errors/warnings when code inspects a query's results to see if that (now-missing) field is/isn't set.

**Reply by drbyte** [reply]
Right. I guess we can leave the db field. How about leaving it defaulted to 1 in db schema, and remove the toggle from the Banner Manager UI?

**Comment by drbyte** [comment]
Done. v2.1.0
